library/L1_Peripheral/lpc40xx/capture.hpp

```diff
@@ -0,0 +1,195 @@
+            .isr = TimerHandler<kTimerPartial3>
+        };
+    };  // struct Channel
+
+
+    static void TimerHandler(const Timer::ChannelPartial_t & channel)
+    {
+        if (*channel.user_callback != nullptr)
+        {
+            (*channel.user_callback)();
+        }
+        channel.timer_register->IR |= 0x30;
+    }
+
+    typedef struct
+    {
+        uint32_t count;
+        uint32_t flags;
+    } CaptureStatus_t;
+
+    static constexpr sjsu::cortex::InterruptController kInterruptController =
+        sjsu::cortex::InterruptController();
+
+    explicit constexpr Capture(
+        const Timer::Channel_t &timer,
+        const sjsu::SystemController &system_controller =
+            DefaultSystemController(),
+        const sjsu::InterruptController &interrupt_controller =
+            kInterruptController
+        )
+        : timer_(timer),
+          system_controller_(system_controller),
+          interrupt_controller_(interrupt_controller)
+    {
+    };
+
+    Status Initialize(
+        units::frequency::hertz_t frequency,
+        IsrPointer isr = nullptr, int32_t priority = -1) const override
+    {
+        // NOTE: Same initialization as Timer library's Initialize()
+        system_controller_.PowerUpPeripheral(timer_.channel.power_id);
+        SJ2_ASSERT_FATAL(
+            frequency.to<uint32_t>() != 0,
+            "Cannot have zero ticks per microsecond, please choose 1 or more.");
+        // Set Prescale register for Prescale Counter to milliseconds
+        uint32_t prescaler =
+            system_controller_
+                .GetPeripheralFrequency(timer_.channel.power_id) / frequency;
+        timer_.channel.timer_register->PR = prescaler;
+        timer_.channel.timer_register->TCR |= (1 << 0);
+        *timer_.channel.user_callback = isr;
+        interrupt_controller_.Register({
+                .interrupt_request_number = timer_.channel.irq,
+                .interrupt_service_routine = timer_.isr,
+                .enable_interrupt = true,
+                .priority = priority
+        });
+
+        // NOTE: New initialization for Capture library
+        ConfigureCapture(kCaptureChannel0, kCaptureEdgeModeNone);
+        ConfigureCapture(kCaptureChannel1, kCaptureEdgeModeNone);
+        EnableCaptureInterrupt(kCaptureChannel0, false);
+        EnableCaptureInterrupt(kCaptureChannel1, false);
+
+        return Status::kSuccess;
+    }
+
+    void GetCaptureStatus(CaptureChannelNumber channel, CaptureStatus_t &status)
+    {
+        status.flags = timer_.channel.timer_register->IR;
+
+        status.count = (channel == kCaptureChannel0)
+            ? timer_.channel.timer_register->CR0
+            : timer_.channel.timer_register->CR1;
+    }
+
+    void ConfigureCapture(
+            CaptureChannelNumber channel,
+            CaptureEdgeMode mode) const override
+    {
+        uint32_t shift = (channel == kCaptureChannel1) ? 3 : 0;
+        timer_.channel.timer_register->CCR &= ~(0x03 << shift);
```

Comment:
Use `util::bit` libraries for bit manipulation. Do not do bit manipulation directly.